interop-tests/src/arch.rs

```diff
@@ -182,16 +191,19 @@ pub(crate) mod wasm {
         ip: &str,
         transport: Transport,
     ) -> Result<(BoxedTransport, String)> {
-        if let Transport::Webtransport = transport {
-            Ok((
+        match transport {
+            Transport::Webtransport => Ok((
                 libp2p::webtransport_websys::Transport::new(
                     libp2p::webtransport_websys::Config::new(&local_key),
                 )
                 .boxed(),
                 format!("/ip4/{ip}/udp/0/quic/webtransport"),
-            ))
-        } else {
-            bail!("Only webtransport supported with wasm")
+            )),
+            Transport::WebRTCWebSys => Ok((
```

Comment:
```suggestion
            Transport::WebRTCDirect => Ok((
```
 I think we can just reuse the existing variant here, right?

Reply:
I thought about that too, but we would have to adjust the logic for dialing and make it more complex here:

https://github.com/DougAnderson444/rust-libp2p/blob/0a8bb1883cb7cb468d83b70e596b9e61a1a282cd/interop-tests/src/lib.rs#L47-L51

Also I thought since's it's the browser we should differentiate it somehow so it's not confusing? 

I don't mind the comments, just trying to keep up with them all =)

Reply:
I will add `#[cfg(not(target_arch = "wasm32"))]` to the line above so Direct only listens when it's not wasm

Reply:
Ah yes, that is a bit convoluted *sigh*

Unfortunately, our _server-side_ WebRTC currently has a limitation where it cannot dial a node without listening first. The linked issue should fix this in which case we would be able to entirely remove that branch and with it the `cfg` you've added.

Long-story short: The `cfg` is fine here, the listening is a workaround here and actually shouldn't be there.

Reply:
Got it, I've inserted the `cfg` for now, we'll remove it when it down the road when it's ready